tree/30_generic_methods/log_rudder.cf

```diff
@@ -141,7 +141,9 @@ bundle agent _log_rudder_v2(expected_reports_source, message, class_prefix, args
 
       # 2/ We remove the last promiser, which is the original call to any of the above logger methods
       "promiser_count"   int => length("promiser_list_nologgerint");
-      "nologger_len"  string => eval("${promiser_count}-1", "math", "infix");
+      "nologger_len_real"  string => eval("${promiser_count}-1", "math", "infix");
+      # Avoid error when casting to int (#7762)
+      "nologger_len" string => format("%d", "${nologger_len_real}");
       "nologger_len_int" int => "${nologger_len}";
```

Comment:
You can remove this line and use directly the string value.
The int was there just for the cast, there is no real int in cfengine, everything is a string